airflow/models/dagbag.py

```diff
@@ -539,6 +539,10 @@ def collect_dags(
 
         # Ensure dag_folder is a str -- it may have been a pathlib.Path
         dag_folder = correct_maybe_zipped(str(dag_folder))
+
+        if not os.path.isfile(dag_folder):
```

Comment:
seems it could alternatively be a folder or None or empty string?

Reply:
Yeah, this should probably be 
```suggestion
        if not os.path.exists(dag_folder):
```

Reply:
In practice I don't think it can. Let me investigate typing.

https://github.com/apache/airflow/blob/26f94c5370587f73ebd935cecf208c6a36bdf9b6/airflow/settings.py#L83

Reply:
@ashb, yeah, definitely should.